src/popup.js

```diff
@@ -28,26 +35,38 @@ const updateSettings = () => {
                 });
         });
     });
+
+    // hide percentages position select if percentages are disabled
+    document.getElementsByClassName('percentagesPosition')[0].style.display = percentages.checked ? 'block' : 'none';
 };
 
 /* Set the initial values for the checkboxes */
 chrome.storage.sync.get(null, (settings) => {
-    if (Object.entries(settings).length === 0 && settings.constructor === Object) { // Enable all functional settings if it hasn't been set yet
-        quickview.checked = true;
-        calculation.checked = true;
-        liveModification.checked = true;
-        percentageColumn.checked = true;
-        updateSettings();
-    } else {
-        quickview.checked = settings.quickview;
-        calculation.checked = settings.calculation;
-        liveModification.checked = settings.liveModification;
-        percentageColumn.checked = settings.percentageColumn;
-    }
+    defaults = {
+        quickview: true,
+        calculation: true,
+        liveModification: true,
+        percentages: true,
+        betterTableLayout: true,
+        percentagesPosition: 'Last column'
+    };
+
+    // assign saved value if it exist, otherwise value from default object above
+    quickview.checked = settings.quickview !== undefined ? settings.quickview : defaults.quickview;
+    calculation.checked = settings.calculation !== undefined ? settings.calculation : defaults.calculation;
+    liveModification.checked = settings.liveModification !== undefined ? settings.liveModification : defaults.liveModification;
+    percentages.checked = settings.percentages !== undefined ? settings.percentages : defaults.percentages;
+    betterTableLayout.checked = settings.betterTableLayout !== undefined ? settings.betterTableLayout : defaults.betterTableLayout;
+    percentagesPosition.value = settings.percentagesPosition !== undefined ? settings.percentagesPosition : defaults.percentagesPosition;
```

Comment:
Can be shortened to:
```suggestion
    quickview.checked = settings.quickview || defaults.quickview;
    calculation.checked = settings.calculation || defaults.calculation;
    liveModification.checked = settings.liveModification || defaults.liveModification;
    percentages.checked = settings.percentages || defaults.percentages;
    betterTableLayout.checked = settings.betterTableLayout || defaults.betterTableLayout;
    percentagesPosition.value = settings.percentagesPosition || defaults.percentagesPosition;
```